runtime/gc_structs/ConstantPoolObjectSlotIterator.cpp

```diff
@@ -35,35 +35,69 @@
 
 
 /**
+ * If condyOnly flag is not set:
  * @return the next object slot in the constant pool
  * @return NULL if there are no more object references
+ *
+ * If condyOnly flag is set and Java version is SE11 and above:
+ * @return the next Condy reference from the constant pool
+ * @return NULL if there are no more references
  */
 j9object_t *
-GC_ConstantPoolObjectSlotIterator::nextSlot()
-{
+GC_ConstantPoolObjectSlotIterator::nextSlot(Boolean condyOnly, MM_EnvironmentBase *env) {
 	U_32 slotType;
 	j9object_t *slotPtr;
 	j9object_t *result = NULL;
-
-	while(_cpEntryCount) {
-		if(0 == _cpDescriptionIndex) {
-			_cpDescription = *_cpDescriptionSlots;
-			_cpDescriptionSlots += 1;
-			_cpDescriptionIndex = J9_CP_DESCRIPTIONS_PER_U32;
+	if (condyOnly) {
+		if (eJ2SE_VERSION((J9JavaVM*) env->getLanguageVM()) < J2SE_V11) {
+			return NULL;
 		}
-
-		slotType = _cpDescription & J9_CP_DESCRIPTION_MASK;
-		slotPtr = _cpEntry;
-
-		/* Adjust the CP slot and description information */
-		_cpEntry = (j9object_t *)( ((U_8 *)_cpEntry) + sizeof(J9RAMConstantPoolItem) );
-		_cpEntryCount -= 1;
-
-		_cpDescription >>= J9_CP_BITS_PER_DESCRIPTION;
-		_cpDescriptionIndex -= 1;
-
-		/* Determine if the slot should be processed */
-		switch (slotType) {
+		while (_cpEntryCount) {
+			if (0 == _cpDescriptionIndex) {
+				_cpDescription = *_cpDescriptionSlots;
+				_cpDescriptionSlots += 1;
+				_cpDescriptionIndex = J9_CP_DESCRIPTIONS_PER_U32;
+			}
+
+			slotType = _cpDescription & J9_CP_DESCRIPTION_MASK;
+			slotPtr = _cpEntry;
+
+			/* Adjust the CP slot and description information */
+			_cpEntry = (j9object_t *) (((U_8 *) _cpEntry)
+					+ sizeof(J9RAMConstantPoolItem));
+			_cpEntryCount -= 1;
+
+			_cpDescription >>= J9_CP_BITS_PER_DESCRIPTION;
+			_cpDescriptionIndex -= 1;
+
+			/* Determine if the slot should be processed */
+			if (slotType == J9CPTYPE_CONSTANT_DYNAMIC) {
+				result = &(((J9RAMConstantDynamicRef *) slotPtr)->value);
+				break;
+			}
+		}
+		return result;
+	} else {
+		while (_cpEntryCount) {
+			if (0 == _cpDescriptionIndex) {
+				_cpDescription = *_cpDescriptionSlots;
+				_cpDescriptionSlots += 1;
+				_cpDescriptionIndex = J9_CP_DESCRIPTIONS_PER_U32;
+			}
+
+			slotType = _cpDescription & J9_CP_DESCRIPTION_MASK;
+			slotPtr = _cpEntry;
+
+			/* Adjust the CP slot and description information */
+			_cpEntry = (j9object_t *) (((U_8 *) _cpEntry)
+					+ sizeof(J9RAMConstantPoolItem));
+			_cpEntryCount -= 1;
+
+			_cpDescription >>= J9_CP_BITS_PER_DESCRIPTION;
+			_cpDescriptionIndex -= 1;
+
+			/* Determine if the slot should be processed */
+			switch (slotType) {
```

Comment:
I think you can recude the amount of code duplication if you put the if around the `switch` instead